backend/app/api/endpoints/devices.py

```
@@ -438,3 +438,100 @@ async def execute_device_command(
+
+    session_id: str = Field(..., description="Unique session identifier")
+    device_id: str = Field(..., description="Target device ID")
+    type: str = Field(..., description="Session type (terminal or code_server)")
+    path: str = Field(..., description="Working directory path")
+    url: str = Field(default="", description="Browser-accessible session URL")
+
+
+@router.post("/{device_id}/terminal", response_model=DeviceSessionResponse)
+async def start_device_terminal(
+    device_id: str,
+    db: Session = Depends(get_db),
```

Ruff (0.15.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)